app/controllers/UserDataController.scala

```diff
@@ -28,18 +31,48 @@ class UserDataController(dynamo: Dynamo, val deps: BaseFaciaControllerComponents
   }
 
   def putFrontIds() = APIAuthAction { request =>
-
-    val frontIds: Option[List[String]] = request.body.asJson.flatMap(jsValue =>
-      jsValue.asOpt[List[String]])
-
-    frontIds match {
-      case Some(ids) => {
-        val userEmail = request.user.email
-        Scanamo.exec(dynamo.client)(userDataTable.update('email -> request.user.email, set('frontIds -> ids)))
+    // We handle both cases here, because of the migration from a list to a map by priority
+    val maybeFrontIds: Option[List[String]] = request.body.asJson.flatMap(
+      _.asOpt[List[String]])
+    val maybeFrontIdsByPriority: Option[Map[String, List[String]]] = request.body.asJson.flatMap(
+      _.asOpt[Map[String, List[String]]])
+    Logger.info(maybeFrontIds.toString)
+    Logger.info(maybeFrontIdsByPriority.toString)
+    (maybeFrontIds, maybeFrontIdsByPriority) match {
+      case (Some(frontIds), _) =>
+        Scanamo.exec(dynamo.client)(userDataTable.update('email -> request.user.email, set('frontIds -> frontIds)))
         Ok
-      }
-      case None => BadRequest
+      case (_, Some(frontIdsByPriority)) =>
+        Scanamo.exec(dynamo.client)(userDataTable.update('email -> request.user.email, set('frontIdsByPriority -> frontIdsByPriority)))
```

Comment:
Just to double check: does the data that we get from the client have editorial fronts marked as having editorial priority instead of empty priority?

Reply:
Good question, and yes, it's explicit 👍